File: Ben/d2/d2.ts

let print = console.log

import * as fs from 'fs';

const data = fs.readFileSync('./d2/input.txt', 'utf-8')
  .split("\n");

function p1(data) {
  let pos_h = 0;
  let pos_d = 0;

  for (let line of data) {
    let line_parts = line.split(' ');
    let cmd = line_parts[0]
    let val = Number(line_parts[1])

    switch (cmd) {
      case 'forward': 
        pos_h += val;
        break;
      case 'down': 
        pos_d += val;
        break;
      case 'up': 
        pos_d -= val;
        break;
    }
  }

  return pos_h * pos_d
}

function p2(data) {
  let pos_h = 0;
  let pos_d = 0;
  let aim = 0;

  for (let line of data) {
    let line_parts = line.split(' ');
    let cmd = line_parts[0]
    let val = Number(line_parts[1])

    switch (cmd) {
      case 'forward': 
        pos_h += val;
        pos_d += aim * val;
        break;
      case 'down': 
        aim += val;
        break;
      case 'up': 
        aim -= val;
        break;
    }
  }

  return pos_h * pos_d
}


print(p1(data))
print(p2(data))

File: Ben/d2/d2.py

data = []
with open('d2/input.txt', 'r') as f:
  data = list(map(lambda a: a.strip(), f.readlines()))
data_len = len(data)

def p1(data):
  pos_h = 0
  pos_d = 0

  for line in data:
    cmd,v = line.split(' ')
    v = int(v)
    if cmd == 'forward':
      pos_h += v
    elif cmd == 'down':
      pos_d += v
    elif cmd == 'up':
      pos_d -= v

  return pos_h * pos_d

def p2(data):
  pos_h = 0
  pos_d = 0
  aim = 0

  for line in data:
    cmd,v = line.split(' ')
    v = int(v)
    if cmd == 'forward':
      pos_h += v
      pos_d += aim * v
    elif cmd == 'down':
      aim += v
    elif cmd == 'up':
      aim -= v

  return pos_h * pos_d

print(p1(data))
print(p2(data))
